gallery_dl/extractor/hatenablog.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://hatenablog.com"""
+
+import re
+from .common import Extractor, Message
+from .. import text
+
+
+BASE_PATTERN = (
+    r"(?:hatenablog:https?://([^/]+)|(?:https?://)?"
+    r"([\w-]+\.(?:hatenablog\.com|hatenablog\.jp"
+    r"|hatenadiary\.com|hateblo\.jp)))"
+)
+QUERY_RE = r"(?:\?([^#]*))?(?:#.*)?$"
+
+
+class HatenablogExtractor(Extractor):
+    """Base class for HatenaBlog extractors"""
+    category = "hatenablog"
+    directory_fmt = ("{category}", "{domain}")
+    filename_fmt = "{category}_{domain}_{entry}_{num:>02}.{extension}"
+    archive_fmt = "{filename}"
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+
+        self.domain = match.group(1) or match.group(2)
+        self._find_img = re.compile(r'<img +(.+?) */?>').finditer
```

Comment:
```suggestion
        self._find_img = re.compile(r'<img +([^>]+)').finditer
```
I generally dislike `.*` in REs and this should work as well.
It might include some extra chars at the beginning and end, but those can be handled otherwise or might not even matter.